bzl/macros/pico.bzl

```diff
@@ -16,54 +17,40 @@ def pico_project(name, srcs, deps, include_host = True):
         name: The name of the project
         srcs: The source files
         deps: The dependencies
-        include_host: Whether to include the host binary (default: True)
     """
-    name = name.replace("_pico", "")
-
-    if include_host:
-        # Host compilation of the binary
-        cc_binary(
-            name = name + "_host",
-            srcs = srcs,
-            deps = deps,
-        )
-
-    # Pico compilation of the binary;
-    # compiled with `--config pico`
-    pico_binary(
-        name = name + ".elf",
+    cc_binary(
+        name = name,
         srcs = srcs,
         deps = deps,
     )
 
-    # Enable stdio over USB
-    pico_build_with_config(
-        name = name + "_usb.elf",
-        input = name + ".elf",
+    # Apply transitions to the `rp2040` configuration
+    rp2040_elf(
```

Comment:
These new rules apply platform transitions, so my embedded code is back to not needing `--config` / `--platforms` flags to compile.